code/sys_shd/src/system_shared_tool/sys_shd_common.py

```diff
@@ -19,6 +25,69 @@
 
 #######################             CLASSES              #######################
 
+class SysShdParamsC:
+    """
+    Class that contains the can parameters in order to create the thread correctly
+    """
+    def __init__(self, target: Callable[..., object] | None = ...,
+        name: str | None = ..., args: Iterable[Any] = ...,
+        kwargs: Mapping[str, Any] | None = ..., *, daemon: bool | None = ...):
+        self.target = target
+        self.name = name
+        self.args = args
+        self.kwargs = kwargs
+        self.daemon = daemon
+
+class SysShdNodeC(Thread):
+    """Classmethod for creating a system shdNode class .
+
+    Args:
+        Thread ([type]): [description]
+    """
+    def __init__(self, cycle_period: float, working_flag : Event,
+                 node_params: SysShdParamsC = SysShdParamsC()) -> None:
+        '''
+        Initialize the thread node.
+        '''
+        super().__init__(group = None, target = node_params.target, name = node_params.name,
+                         args = node_params.args, kwargs = node_params.kwargs,
+                         daemon = node_params.daemon)
+        self.working_flag: Event = working_flag
+        self.cycle_period: float = cycle_period
+
+    def sync_shd_data(self) -> None:
+        """Function to be implemented in the inherited class
+        """
+
+    def process_iteration(self) -> None:
+        """Function to be implemented in the inherited class
+        """
+
+    def stop(self) -> None:
+        """
+        Stop the node thread .
+        """
+        log.critical("Stopping node thread.")
+        self.join()
```

Comment:
Add try, except if node is down

Reply:
self.join() doesn´t make sense here as the action can not be done in the current node